test_pull_requests

```diff
@@ -2323,9 +2341,19 @@ elsif local_merge_pull_id
   @api_client.local_merge_pull_request(local_merge_pull_id, pull_id_repo)
 elsif test_merge_pull_id
   @api_client.test_merge_pull_request(test_merge_pull_id, pull_id_repo, Properties['settings']['merge_test_settings'])
+elsif rate_limit
+  puts "Remaining: #{@api_client.rate_limit_remaining}"
+  puts "Resets in: #{@api_client.rate_limit_reset.to_i - Time.now.to_i}s, at #{Time.at(@api_client.rate_limit_reset.to_i).to_datetime} (#{@api_client.rate_limit_reset})"
 else
   # Process all the pull requests for testing
   $stderr.puts "Processing pull requests..."
-  @api_client.process_pull_requests(merge_pretest_success)
+  $stderr.puts "Rate limit remaining: #{@api_client.rate_limit_remaining}"
+  starting_limit = @api_client.rate_limit_remaining
+  if @api_client.rate_limit_remaining.to_i < 1000
```

Comment:
We back off pretty often, it seems like. I updated the commit with logic that will do the backoff unless the rate limit reset is pretty soon (within 8 minutes, constant `JOB_INTERVAL`)